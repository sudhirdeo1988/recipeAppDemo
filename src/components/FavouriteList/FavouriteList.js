import React, { useState } from "react";
import { connect } from "react-redux";
import { FAV_RECIPES } from "../../utilities/constants";
import Modal from "react-bootstrap/Modal";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import "./FavouriteList.scss";

function FavouriteList(props) {
  const { favRecipe } = props;
  const [show, setShow] = useState(false);

  const removeItem = (item) => {
    props.removeFromFavRecipe(item);
  };
  const handleClose = () => {
    setShow(false);
  };
  const handleShow = () => {
    setShow(true);
  };
  const count = favRecipe.length;

  return (
    <div className="c-favouriteList">
      <button className="listBtn" onClick={handleShow}>
        <i className="icon ts-heart-o"></i>
        <span className="itemCount">{count}</span>
      </button>

      <Modal show={show} onHide={() => handleClose()} animation={false}>
        <Modal.Header closeButton>
          <Modal.Title>Modal heading</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div className="clearfix">
            {!count ? (
              <span>No records</span>
            ) : (
              favRecipe.map((recipe) => {
                return (
                  <div className="recipeFavList" key={recipe.id}>
                    <img
                      className="img"
                      src={recipe.image}
                      alt={recipe.label}
                    />
                    <button
                      type="button"
                      className="remoceBtn"
                      onClick={() => removeItem(recipe)}
                    ></button>
                    <div className="recipeTitle">{recipe.label}</div>
                    <div className="recipeInfo">
                      <Row>
                        <Col xs={{ span: 6 }}>
                          <label>Calories: </label>
                          <span>{recipe.calories.toFixed(2)}&nbsp;kcal</span>
                        </Col>
                        <Col xs={{ span: 6 }}>
                          <label>Total Weight: </label>
                          <span>{recipe.totalWeight.toFixed(2)}&nbsp;g</span>
                        </Col>
                      </Row>
                    </div>
                  </div>
                );
              })
            )}
          </div>
        </Modal.Body>
      </Modal>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    favRecipe: state.favouriateRecipe,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    removeFromFavRecipe: (payload) =>
      dispatch({ type: FAV_RECIPES.REMOVE.RECIPEITEM, payload }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(FavouriteList);
