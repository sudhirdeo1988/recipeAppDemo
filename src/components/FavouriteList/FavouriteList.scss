@import '../../assets/scss/utilities.scss';

.c-favouriteList{
  position: absolute;
  height: 40px;
  width: 40px;
  top: 40px;
  right: 40px;
  z-index: 4;

    .listBtn{
      background-color: transparent;
      border: 0;
      display: block;
      height: 40px;
      width: 40px;
      padding: 0;
      outline: none;

        .icon{
          color: #fff;
          display: block;
          font-size: 38px;
          line-height: 32px;
        }
        .itemCount{
          background-color: #fff;
          position: relative;
          padding: 6px 3px 5px;
          font-size: 10px;
          font-weight: 600;
          color: $primaryColor;
          line-height: 7px;
          display: inline-block;
          border-radius: 30px;
          top: -46px;
          right: -17px;
          height: 20px;
          width: 20px;
          box-sizing: border-box;
          text-align: center;
        }
    }
}
.modal{
  font-family: 'Roboto Condensed', sans-serif;
  .modal-dialog{
      .modal-content{
        border-top: 5px solid $primaryColor;

          .modal-header{
            padding: 10px;
              .modal-title{
                color: $primaryColor;
                font-size: 18px;
                font-weight: 600;
                text-transform: uppercase;
                line-height: 22px;
              }
          }
          .modal-body{
            padding: 10px;
          }
      }
  }
}
.recipeFavList{
  padding: 5px 30px 5px 125px;
  min-height: 80px;
  border: 1px solid #e6e6e6;
  background-color: #f9f9f9;
  margin-bottom: 20px;
  position: relative;

    .img{
      display: block;
      position: absolute;
      height: 67px;
      width: 110px;
      top: 5px;
      left: 5px;
    }
    .remoceBtn{
      padding: 0;
      display: block;
      position: absolute;
      height: 20px;
      width: 20px;
      top: 5px;
      right: 5px;
      background-color: transparent;
      border-radius: 20px;
      border: 0;
      outline: none;
        &::before{
          font-family: "themestudio";
          display: block;
          color: #999;
          content: "\66";
          font-size: 14px;
          line-height: 14px;
        }
    }
    .recipeTitle {
      display: block;
      font-size: 16px;
      line-height: 22px;
      font-weight: 600;
      color: #D91488;
      width: 100%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
  }
    .recipeInfo{
      border-top: 1px solid #eee;
      margin: 5px 0 0;
      padding: 5px 0 0;

        label{
          display: block;
          color: #cacaca;
          display: block;
          font-size: 12px;
          line-height: 14px;
        }
        span{
          display: block;
          color: #999;
          display: block;
          font-size: 12px;
          font-weight: 600;
          line-height: 18px;
        }
    }
}
