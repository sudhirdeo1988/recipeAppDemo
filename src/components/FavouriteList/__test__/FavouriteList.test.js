import React from "react";
import { render, cleanup } from "@testing-library/react";
import configureMockStore from "redux-mock-store";
import thunkMiddleware from "redux-thunk";
import { Provider } from "react-redux";
import FavouriteList from "../FavouriteList";

afterEach(cleanup);
describe("Favourite List Componets", () => {
  test("should render correctly", () => {
    const initialState = {
      favouriateRecipe: [
        {
          uri:
            "http://www.edamam.com/ontologies/edamam.owl#recipe_09be6cc615db3ab88b48dadc30726fed",
          label: "All-Butter Pie Crust",
          image:
            "https://www.edamam.com/web-img/b50/b5031e0c62073138c6b2f878430337f7.JPG",
          source: "Food52",
          url: "https://food52.com/recipes/6718-all-butter-pie-crust",
          shareAs:
            "http://www.edamam.com/recipe/all-butter-pie-crust-09be6cc615db3ab88b48dadc30726fed/all",
          yield: 2,
          calories: 2651.34,
          totalWeight: 645.0861952394988,
          totalTime: 62,
          id: "4cc6e30-effb-5d6b-8b61-8d8f5b54e525",
        }
      ],
    };
    const mockStore = configureMockStore([thunkMiddleware]);
    const store = mockStore(initialState);

    const { container } = render(
      <Provider store={store}>
        <FavouriteList />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
