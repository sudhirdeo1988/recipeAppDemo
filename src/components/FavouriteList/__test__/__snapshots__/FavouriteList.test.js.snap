// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourite List Componets should render correctly 1`] = `
<div>
  <div
    class="c-favouriteList"
  >
    <button
      class="listBtn"
    >
      <i
        class="icon ts-heart-o"
      />
      <span
        class="itemCount"
      >
        1
      </span>
    </button>
  </div>
</div>
`;
